### PR TITLE
allow migration to Keycard keeping the db

### LINKED ISSUE #11659: Keycard migration flow is missing the popup that informs user about migration success
# Bug Report

## Problem
https://github.com/status-im/status-react/pull/11526#issuecomment-762736743

#### Expected behavior
Designs in comment linked above.

### LINKED ISSUE #11777: Enable keeping database when migrating account to keycard
relates to #11431 "Migrate account to a regular keycard from logged out "Manage keys and storage"

When solving #11431 we intentionally left out the case where the user wants to keep its data base when migrating. So the user is forced to reset its database and the check box is checked and greyed out in flow like this https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=8218%3A7450

This issue is about enabling the user to keep its database
- having "reset database" unchecked by default 
- letting the user checks it or not 

Figma design 
https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=8218%3A8089

### PR DESCRIPTION
**Pull request by bitgamma**
Fixes #11777.
Fixes #11659.

- [x] Convert the account from regular to Keycard
- [x] Login automatically immediately after conversion
- [x] Implement password prompt/verification
- [x] Add migration success message after the (automatic) login
- [x] Add error handling
- [x] Create status-go release (to do right before merging)

### CODE REVIEW

File: src/status_im/ui/screens/multiaccounts/key_storage/views.cljs [hunk 1]
```diff
@@ -195,9 +195,37 @@
...
+(defview migrate-account-password-view []
+  (letsubs [{:keys [migration-password-error migration-password-valid?]} [:keycard]]
+    [react/view {:flex 1}
+     [react/view styles/header
+      [react/text {:style {:typography :title-bold}} (i18n/label :t/enter-password)]
+      [react/view {:padding-horizontal 24}
+       [quo/button {:type :secondary
+                    :on-press #(re-frame/dispatch [::multiaccounts.key-storage/skip-password-pressed])}
+        (i18n/label :t/skip)]]]
+     [react/view {:height 1 :background-color colors/gray-lighter}]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/multiaccounts/key_storage/views.cljs]: is this quo/separator ?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/multiaccounts/key_storage/views.cljs (hunk 1 above)
Comment (bitgamma) [reply src/status_im/ui/screens/multiaccounts/key_storage/views.cljs]: indeed. Fixed, thanks!

**Comment by bitgamma** [comment]
failed test is because the reset database checkbox is not checked anymore automatically when migrating (by default, we want to keep the db).

**Comment by churik** [comment]
@bitgamma thank you! Works great.
I'll rework e2e.
Several issues:
### ISSUE 1: account restored from seed phrase (or private key) remains after moving to keycard storage

Currently, if you create multiaccount with keycard, there is no option to add account in wallet using seed phrase or private key. 
But if it was added before migration it is preserved and basically, it is not working (i.e. can't sign transactions)
**Steps:**
1) restore multiaccount from seed phrase
2) add another account in wallet, i.e. 'Added'(using seed phrase or private key)
3) log out and migrate multiaccount to keycard storage, keeping database
4) try to sign transaction from 'Added'

**Expected result:**
can't see 'Added' in accounts list (and will be aware of it during reset)
**Actual result:**
at attempt to sign transaction 
IMAGE 2021-07-19 15:35:37

**OS:** IOS, Android

### ISSUE 2: 'Lock app with fingerprint (Face ID)' remains enabled after moving to keycard storage

IMO better to reset it and ask user to enable it again, as currently it crashes app on IOS and not working until app reopening (religion) on Android

**Steps:**
1) restore multiaccount from seed phrase
2) go to Profile > Privacy and Security > Lock app with fingerprint > enable it
3) migrate multiaccount to keycard storage (with enabled "Save password" on login screen) , keeping database

**Expected result:**
Will be turned off after login
**Actual result:**
- IOS: crash after migrating; app is working after relogin
- Android: Lock app with fingerprint (Face ID)'  enabled, but it doesn't work until relogin to app

**Comment by bitgamma** [comment]
@churik thanks for catching this! should be fixed now. Seed and private keys subaccounts are removed, while watch and derived paths are kept.

**Comment by churik** [comment]
@bitgamma previous issues are fixed, thank you!
Found another one, that may be missed yesterday.
### ISSUE 3: "back up seed phrase" is shown  if migrate restored multiaccount to keycard

**Steps:**
1) restore multiaccount from seed phrase
2) migrate multiaccount to keycard storage

**Expected result:**
no "back up seed phrase" is shown

**Actual result:**
"back up seed phrase" is shown, and error appears if you start process
IMAGE 2021-07-20 11:02:53

**OS:** IOS, Android
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/6847000/Status-debug-logs.zip)

**Comment by bitgamma** [comment]
@churik fixed now, thanks!

**Comment by churik** [comment]
All fixed.
Checked that migrated:
- all types of chats
- browser bookmarks
- different account in wallet
- settings in Profile
- status

**Pull request merged**